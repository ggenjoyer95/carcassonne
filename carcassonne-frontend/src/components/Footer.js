function Footer() {
  return (
    <footer
      style={{
        backgroundColor: "#f5f5f5",
        padding: "10px",
        marginTop: "20px",
        textAlign: "center",
      }}
    >
      <p>Â© 2025 Carcassonne.</p>
    </footer>
  );
}

export default Footer;
