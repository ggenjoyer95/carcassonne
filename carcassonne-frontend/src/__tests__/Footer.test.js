import React from "react";
import { render, screen } from "@testing-library/react";
import Footer from "../components/Footer";

describe("Footer component", () => {
  test("renders without crashing and shows correct text", () => {
    render(<Footer />);

    const textElement = screen.getByText(/Â© 2025 Carcassonne\./i);

    expect(textElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
