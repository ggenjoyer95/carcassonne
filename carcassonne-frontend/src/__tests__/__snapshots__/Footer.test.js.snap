// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    style="background-color: rgb(245, 245, 245); padding: 10px; margin-top: 20px; text-align: center;"
  >
    <p>
      Â© 2025 Carcassonne.
    </p>
  </footer>
</DocumentFragment>
`;
